d3rlpy/algos/qlearning/torch/callbacks.py

```diff
@@ -15,27 +16,45 @@ def __call__(self, algo: QLearningAlgoBase, epoch: int, total_step: int):
 
 
 class ParameterReset(QLearningCallback):
-    def __init__(self, replay_ratio: int, layer_reset:Sequence[bool], 
-                 algo:QLearningAlgoBase=None) -> None:
+    def __init__(self, replay_ratio: int, encoder_reset:Sequence[bool],
+                 output_reset:bool, algo:QLearningAlgoBase=None) -> None:
         self._replay_ratio = replay_ratio
-        self._layer_reset = layer_reset
+        self._encoder_reset = encoder_reset
+        self._output_reset = output_reset
         self._check = False
         if algo is not None:
             self._check_layer_resets(algo=algo)
             
     
+    def _get_layers(self, q_func:nn.ModuleList)->List[nn.Module]:
+        all_modules = {nm:module for (nm, module) in q_func.named_modules()}
+        q_func_layers = [
+            *all_modules["_encoder._layers"],
```

Comment:
@takuseno assuming you're happy with the general approach of using the epoch_callback to inject the parameter reset functionality - I wondered if you could recommend a better approach for obtaining the encoder and fc layers which follows static typing?